scalyr_agent/json_lib/parser.py

```diff
@@ -290,6 +288,9 @@ def __parse_object(self):
             else:
                 return self.__error("Expected string literal for object attribute name")
 
+            if key is not None:
```

Comment:
Why do you need this?  Can you add a comment

Reply:
`key` variable can be `None` after` self.__parse_string()'.

https://github.com/scalyr/scalyr-agent-2/blob/c4bfd11c7e44597b66d81d0d5f2727336a135450/scalyr_agent/json_lib/parser.py#L462-L470

but now I can see that an additional check for a quote does not make much sense and can be removed at all. Do you agree?

Reply:
Well, it should raise an error instead.  It's probably a good idea to verify preconditions.

Reply:
Done. 

I just wanted to say, there is already verification for a double quote before every '__parse_string' function call and it just wouldn’t have been called in the opposite case.  But now I agree with you because this additional check can prevent errors in the future.